src/client/common/insidersBuild/insidersExtensionService.ts

```diff
@@ -28,7 +28,10 @@ export class InsidersExtensionService implements IExtensionSingleActivationServi
     public async activate() {
         this.registerCommandsAndHandlers();
         const installChannel = this.extensionChannelService.getChannel();
-        await this.handleEdgeCases(installChannel);
+        if (await this.handleEdgeCases(installChannel)) {

```

Comment:
FWIW, sometimes it's a little clearer to use a well-named intermediate variable:

```suggestion
        const alreadyHandled = await this.handleEdgeCases(installChannel);
        if (alreadyHandled) {
```